src/math_funcs.jl

```diff
@@ -0,0 +1,77 @@
+# define maths function for rotations types
+import Base.(*), Base.(.*), Base.(-)
+
+function add_maths(rot_type)
+
+    # create an expression for a tuple X[1], X[2], etc...
+    fields = fieldnames(rot_type)
+
+    # the code block we are creating
+    qb = quote; end
+
+    # allow rotating a Vec{3, ...} using *    
+    if !(rot_type <: FixedArray)
+        qn = quote
+            (*){T <: Real}(R::$(rot_type), X::Vec{3,T}) = rotate_point(R, X)
+        end
+        append!(qb.args, qn.args)
+    end
+    
+
+    # allow multiplcation by a scalar (might make sense with small angle approximations)
```

Comment:
Hmm... is this a functionality we want exported? Should people do dangerous stuff by hand? I'm worried we could get strange bugs out of this... plus it doesn't work the same way for `Quaternion`...